### PR TITLE
#1104 Modify docs for background tasks

### PR DESCRIPTION
**Pull request by f0t0n**
## What do these changes do?

Changes:

- Modify docs for background tasks
- Add an example code to examples directory

As discussed the old code example for background tasks creating was working but not optimal.
Removed `asyncio.gather()` from the example and used `app.loop.create_task()` instead.
Also added an example of tasks cancellation on cleanup.

## Are there changes in behavior for the user?

None

## Related issue number

#1104

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes

### CODE REVIEW

**Comment by f0t0n** [comment]
@asvetlov Do you think it's short enough or I should remove something (https://github.com/KeepSafe/aiohttp/issues/1104#issuecomment-241540140)?

image

**Comment by asvetlov** [comment]
First.
I think `quickly_notify_monitoring` is not necessary.
If you want to inform users to allowance for doing an instant activity on app startup -- please invite something more obvious than just `sleep`.
I have troubles with a good example personally,

Second.
Your code uses both zmq and redis. For sake of shortness we should choose the single one.
I believe redis is more well known library, let's keep it.

Redis example may be more realistic than, again, just sleeps in a loop.
Create and redis connection using aioredis, subscribe for events and iterate over subscribed messages.

I gives to reader a clue for real use cases.

**Comment by f0t0n** [comment]
@asvetlov Thanks for guidance. I've just pushed new changes and rebased the branch to lastest master.

Currently it looks like this:

image

Also I've rewrote examples/background_tasks.py and ran it with redis to ensure that it works correctly. So the example in docs is an extraction from there and is working fine.

File: examples/background_tasks.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+async def listen_to_redis(app):
+    try:
+        sub = await aioredis.create_redis(('localhost', 6379), loop=app.loop)
+        ch, *_ = await sub.subscribe('news')
+        async for msg in ch.iter(encoding='utf-8'):
+            # Forward message to all connected websockets:
+            for ws in app['websockets']:
+                ws.send_str('{}: {}'.format(ch.name, msg))
+            print("message in {}: {}".format(ch.name, msg))
+    except asyncio.CancelledError:
```
Comment (asvetlov) [inline examples/background_tasks.py:34]: `finally` is better than `except CancelledError`.
We should close redis subscription unconditionally.

Comment (f0t0n) [reply examples/background_tasks.py:34]: Agreed. Gonna modify now.

Comment (f0t0n) [reply examples/background_tasks.py:34]: Done.

**Comment by f0t0n** [comment]
@asvetlov Should I squash commits or keep for history?

**Comment by asvetlov** [comment]
Doesn't matter -- I'm doing squashing on merge anyway.

**Comment by asvetlov** [comment]
Perfect!

**Pull request merged**